arquillian/resolver/src/main/java/org/wildfly/swarm/arquillian/resolver/ShrinkwrapArtifactResolvingHelper.java

```diff
@@ -110,16 +118,48 @@ public ArtifactSpec resolve(ArtifactSpec spec) {
     }
 
     @Override
-    public Set<ArtifactSpec> resolveAll(final Set<ArtifactSpec> specs) {
+    public Set<ArtifactSpec> resolveAll(final Set<ArtifactSpec> specs, boolean trasitive) {
         if (specs.isEmpty()) {
-
             return specs;
         }
 
+        MavenResolutionStrategy transitivityStrategy = ( trasitive ? TransitiveStrategy.INSTANCE : NonTransitiveStrategy.INSTANCE );
+
         resetListeners();
         final MavenResolvedArtifact[] artifacts =
                 withResolver(r -> r.resolve(specs.stream().map(ArtifactSpec::mavenGav).collect(Collectors.toList()))
-                        .withTransitivity()
+                        .using(new CombinedStrategy(
+                                new MavenResolutionStrategy() {
+                                    @Override
+                                    public TransitiveExclusionPolicy getTransitiveExclusionPolicy() {
+                                        return null;
+                                    }
+
+                                    @Override
+                                    public MavenResolutionFilter[] getResolutionFilters() {
+                                        return new MavenResolutionFilter[]{
+                                                new MavenResolutionFilter() {
+                                                    @Override
+                                                    public boolean accepts(MavenDependency dependency, List<MavenDependency> dependenciesForResolution, List<MavenDependency> dependencyAncestors) {
+                                                        if ( dependency.getVersion().equals( "1.0.1B" ) ) {
```

Comment:
Ah dang it. Yes.

On Wednesday, August 31, 2016, Ken Finnigan <notifications@github.com>
wrote:

> In arquillian/resolver/src/main/java/org/wildfly/swarm/
> arquillian/resolver/ShrinkwrapArtifactResolvingHelper.java
> <https://github.com/wildfly-swarm/wildfly-swarm/pull/123#discussion_r77049751>
> :
>
> >          resetListeners();
> >          final MavenResolvedArtifact[] artifacts =
> >                  withResolver(r -> r.resolve(specs.stream().map(ArtifactSpec::mavenGav).collect(Collectors.toList()))
> > -                        .withTransitivity()
> > +                        .using(new CombinedStrategy(
> > +                                new MavenResolutionStrategy() {
> > +                                    @Override
> > +                                    public TransitiveExclusionPolicy getTransitiveExclusionPolicy() {
> > +                                        return null;
> > +                                    }
> > +
> > +                                    @Override
> > +                                    public MavenResolutionFilter[] getResolutionFilters() {
> > +                                        return new MavenResolutionFilter[]{
> > +                                                new MavenResolutionFilter() {
> > +                                                    @Override
> > +                                                    public boolean accepts(MavenDependency dependency, List<MavenDependency> dependenciesForResolution, List<MavenDependency> dependencyAncestors) {
> > +                                                        if ( dependency.getVersion().equals( "1.0.1B" ) ) {
>
> Are these printlns leftover from testing?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wildfly-swarm/wildfly-swarm/pull/123/files/3c19dcf3389daaf0060cc7169d7cc361c47fbb8e#r77049751>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA-T61zu1HRGAgeiF9tfYRU-vxIrl-Eks5qlc1EgaJpZM4Jx3g7>
> .
>